trestlebot/tasks/regenerate_task.py

```diff
@@ -36,27 +36,23 @@ class RegenerateTask(TaskBase):
 
     def __init__(
         self,
-        working_dir: str,
-        authored_model: str,
+        authored_object: AuthoredObjectBase,
         markdown_dir: str,
-        ssp_index_path: str = "",
         model_filter: Optional[ModelFilter] = None,
     ) -> None:
         """
         Initialize regenerate task.
 
         Args:
-            working_dir: Working directory to complete operations in
-            authored_model: String representation of model type
+            authored_object: Object that can regenerate Markdown content from JSON
             markdown_dir: Location of directory to write Markdown in
-            ssp_index_path: Path of ssp index JSON in the workspace
             model_filter: Optional filter to apply to the task to include or exclude models
             from processing.
         """
 
-        self._authored_model = authored_model
+        self._authored_object = authored_object
         self._markdown_dir = markdown_dir
-        self._ssp_index_path = ssp_index_path
+        working_dir = self._authored_object.get_trestle_root()
```

Comment:
I did not see a check (ex. os.path.exists) anywhere validating this ... maybe here is not a good place but thought I would mention considering some guards here.

Reply:
Good point. Seems like a good addition to the flag input validation to get that feedback the end-user faster.